plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/S3FileSystemExchangeStorage.java

```diff
@@ -107,7 +107,7 @@
 public class S3FileSystemExchangeStorage
         implements FileSystemExchangeStorage
 {
-    private static final String DIRECTORY_SUFFIX = "_$folder$";
+    private static final String GCS_SCHEME = "gs";
```

Comment:
I would rather determine operation mode when we setup `S3FileSystemExchangeStorage` instead of looking at schema of directory here and there.

You can do that by defining enum which denotes operation mode and pass the value to constructor:
```java

    public enum CompatibilityMode {
        AWS,
        GCP
    }

    @Inject
    public S3FileSystemExchangeStorage(S3FileSystemExchangeStorageStats stats, ExchangeS3Config config, CompatibilityMode compatibilityMode)
```

And then in `FileSystemExchangeModule` you can do:
```java
            S3FileSystemExchangeStorage.CompatibilityMode compatibilityMode = scheme.equals("gs") ? GCP : AWS;
            binder.bind(S3FileSystemExchangeStorage.CompatibilityMode.class).toInstance(compatibilityMode);
```